pkg/devspace/configure/image.go

```diff
@@ -136,65 +137,73 @@ func (m *manager) AddImage(imageName, image, dockerfile, contextPath string, doc
-			useGithubRegistry = useGithubRegistry + fmt.Sprintf(registryUsernameHint, authConfig.Username)
+		if !isLoggedIntoRegistry {
+			authConfig, err = dockerClient.GetAuthConfig(githubContainerRegistry, true)
+			if err == nil && authConfig.Username != "" {
+				useGithubRegistry = useGithubRegistry + fmt.Sprintf(registryUsernameHint, authConfig.Username)
+				registryDefaultOption = useGithubRegistry
+				registryOptions = []string{useGithubRegistry, useDockerHub, useOtherRegistry, skipRegistry}
+				isLoggedIntoRegistry = true
+			}
 		}
 
-		registryOptions := []string{useDockerHub, useGithubRegistry, useOtherRegistry}
 		selectedRegistry, err := m.log.Question(&survey.QuestionOptions{
-			Question:     "Which registry would you want to use to push images to? (optional, choose any)",
+			Question:     "Which registry would you want to use to push images to?",
 			DefaultValue: registryDefaultOption,
 			Options:      registryOptions,
 		})
 		if err != nil {
 			return err
 		}
 
-		registryHostname := ""
+		if selectedRegistry != skipRegistry {
+			registryHostname := ""
+			// if selectedRegistry == useDockerHub do nothing.
+			if selectedRegistry == useGithubRegistry {
+				registryHostname = githubContainerRegistry
+			} else {
+				registryHostname, err = m.log.Question(&survey.QuestionOptions{
+					Question:     "Please provide the registry hostname without the image path (e.g. gcr.io, ghcr.io, ecr.io)",
+					DefaultValue: "gcr.io",
+				})
+				if err != nil {
+					return err
+				}
+			}
 
-		if selectedRegistry == useDockerHub {
-			// nothing to do here
-		} else if selectedRegistry == useGithubRegistry {
-			registryHostname = githubContainerRegistry
-		} else {
-			registryHostname, err = m.log.Question(&survey.QuestionOptions{
-				Question:     "Please provide the registry hostname without the image path (e.g. gcr.io, ghcr.io, ecr.io)",
-				DefaultValue: "gcr.io",
-			})
+			registryUsername, err := m.addPullSecretConfig(dockerClient, strings.Trim(registryHostname+"/test/test", "/"))
 			if err != nil {
 				return err
 			}
-		}
 
-		registryUsername, err := m.addPullSecretConfig(dockerClient, strings.Trim(registryHostname+"/test/test", "/"))
-		if err != nil {
-			return err
-		}
-
-		if registryUsername == "" {
-			registryUsername = "username"
-		}
+			if registryUsername == "" {
+				registryUsername = "username"
+			}
 
-		if selectedRegistry == useDockerHub {
-			imageConfig.Image = registryUsername + "/" + imageName
-		} else {
-			projectPath := registryUsername + "/project"
-			if regexp.MustCompile("^(.+\\.)?gcr.io$").Match([]byte(registryHostname)) {
-				project, err := exec.Command("gcloud", "config", "get-value", "project").Output()
+			if selectedRegistry == useDockerHub {
+				imageConfig.Image = registryUsername + "/" + imageName
+			} else {
+				projectPath := registryUsername + "/project"
+				if regexp.MustCompile(`^(.+\\.)?gcr.io$`).Match([]byte(registryHostname)) {
```

Comment:
Is this correct? As `"\\"` != `` `\\` ``

Reply:
We don't need to escape backslash when we use raw string.
It is recommended to use a raw string in regex. https://staticcheck.io/docs/checks#S1007

Reply:
Removed the extra \.